### PR TITLE
add candid charm

### PR DESCRIPTION
**Pull request by rogpeppe**
We also adjust the rest of the code to work better with the charm.
The charm uses on a resource containing a compressed tarball
with the binary and the other assets required by the service.

Other changes:

- change the release mechanism in the Makefile to work with the
new charm resource scheme.

- remove the possibility of configuring the admin username (it's always "admin@idm" now).
- remove the possibility of changing the HTTP server port (it's always 8080 now).
- add a rendezvous-timeout config option and remove the old unused wait-timeout option.
- remove admin-password as a required configuration option - if there's no admin password, we don't allow basic-auth login.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/13/

File: charm/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CHARM?=/snap/bin/charm
+CHARM_HELPERS_SYNC?=./charm_helpers_sync.py
```
Comment (mhilton) [inline charm/Makefile]: I don't think we need a variable for this now.

File: cmd/candidsrv/main.go [hunk 2]
```diff
@@ -38,43 +38,49 @@ import (
...
+	if err := serve(conf); err != nil {
+		fmt.Fprintf(os.Stderr, "STOP %v\n", err)
+		exit(1)
+	}
+	fmt.Fprintln(os.Stderr, "STOP no error, weirdly")
+	exit(0)
+}
+
+// exit calls os.Exit, first sleeping for a bit to work
+// around an outrageous shitstemd bug which causes
```
Comment (mhilton) [inline cmd/candidsrv/main.go]: let's not have vulgar comments please.

File: server.go [hunk 3]
```diff
@@ -105,9 +98,11 @@ type ServerParams struct {
...
 	DebugStatusCheckerFuncs []debugstatus.CheckerFunc
 
-	// WaitTimeout holds the time after which an interactive discharge wait
-	// request will timeout.
-	WaitTimeout time.Duration
+	// RendezvousTimeout holds the time after which an interactive discharge wait
+	// request will time out.
+	RendezvousTimeout time.Duration
+
+	DebugTeams []string // TODO remove me!
```
Comment (mhilton) [inline server.go]: Why move this?

Comment (rogpeppe) [reply server.go]: I've put it back to where it was for the time being.

File: cmd/candidsrv/main.go (hunk 2 above)
Comment (rogpeppe) [reply cmd/candidsrv/main.go]: aw, done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/14/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**